ostd/src/mm/vm_space.rs

```diff
@@ -97,9 +109,39 @@ impl VmSpace {
         Ok(self.pt.cursor_mut(va).map(CursorMut)?)
     }
 
-    /// Activates the page table.
+    /// Activates the page table on the current CPU.
     pub(crate) fn activate(&self) {
-        self.pt.activate();
+        let _preempt_guard = disable_preempt();
+
+        let mut activated_cpus = self.activated_cpus.lock();
```

Comment:
It just occurs to me that it would be useful for `SpinLockGuard` to provide a getter method for its IRQ or preemption disabling guard. This will benefit CPU local accessor methods that can take these guards.

Reply:
Yes, wanted for it several days ago.